File: Algorithm/binarySearch.py
data = [1,2,3,4,5,6,7,8,9,10]


def binary_search(target, data):
    start = 0
    end = len(data) - 1

    while start <= end:
        mid = (start + end) // 2 # 자료 반띵

        if data[mid] == target: # 중간값과 비교하기
            return mid
        elif data[mid] < target:
            start = mid + 1
        else:
            end = mid -1

    return None

print(binary_search(3, data))

File: Algorithm/binary_search.go
// binary search in a sorted list

func BinarySearch(data []int, value int) bool{

  size := len(data)
  var mid int
  low := 0
  high := size-1
  
  for low <= high{
    mid = low+(high-low) / 2
    if data[mid] == value {
      return true
    } else{
      if data[mid] < value {
        low = mid +1
      } else {
        high = mid -1
      }
    }
  }
return false
}
